async-nats/tests/client_tests.rs

```diff
@@ -535,6 +535,46 @@ mod client {
         rx.recv().await;
     }
 
+    #[tokio::test]
+    async fn connect_timeout() {
```

Comment:
Didn't get into details of the test, but the Client still connects to the TcpSocket and hangs indefinitely on my machine, because it waits for OP from the server (`let op = connection.read_op().await?;` in `try_connect_to` fn.

That makes me wonder - we probably should timeout all - the tcp connection and, tls, and the whole `try_connect_to`.

Additional timeouts are out of scope for this PR though.

Reply:
i have an old macbook lying around. i'll try and fix the test on it. but as you mention - i've already noticed a few other places missing a timeout. would it make more sense to rename the "connetion_timeout" field to "timeout" or "operation_timeout"? and then it can be propogated/used in other operations as well?

in my own code/libs i usually have the read_op as internal_read_op with no timeout, and only available inside the crate, and the 2 methods exposed are read_op_for which accepts a duration for timeout and wraps a read_op_internal and a read_op which calls read_op_for with a hardcoded/configurable sensible timeout - that are both exposed. that way, api compatibility is kept (not broken) and a customizable api is available per operation without the need to have a settable duration field for each operation.

i could add that as another pull request if it helps, but i'm not sure if this is something you want to discuss/develop internally?

Reply:
We definitely want to improve resilience and responsiveness of the library, but that needs a little more thought, especially in asynchronous system like NATS. Most use cases should be handled by ping/pong mismatches and timeouts, but I will look into that.

For this PR, let's focus on connect timeout.